packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -2204,6 +2637,9 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin {
 
   @override
   void didChangeDependencies() {
+    _scaffoldMessenger = ScaffoldMessenger.of(context);
```

Comment:
What happens if we're rebuilt under a different ScaffoldMessenger? Don't we need to unregister with the old one?

Reply:
Added handling.